### PR TITLE
V8: Split IndexCreator and Index config methods

### PR DESCRIPTION
**Pull request by bielu**
I notice IndexCreator actually has 2 responsibilities, which in my opinion is not the best way of doing that and I think it should be separate as in that case it looks weird: 
I want override index config I have to override whole umbracoindexcreator.

And here I have an additional case which could be useful:
I am preparing the package which creates elasticsearch indexes and second developer @callumbwhyte  have a package which wants override config in examine and actually we are hitting here wall if we want install both packages as both cases overriding umbracoindexcreator. 

In a way what I am proposing we could have both packages without stress about the ordering of components and handling differently situation depends on installed packages.

### CODE REVIEW

**Comment by Shazwazza** [comment]
Yep that's great! I haven't had a chance to test but a peek at the code looks good.... But we can't have breaking changes so you can't remove those methods, your gonna have to obsolete them and wrap their response.

**Comment by bielu** [comment]
@Shazwazza ok I change that to have them and actually, I didn't remove them and move them but ok made them Obsolete and wrap answer from IUmbracoIndexConfig as they are just in a different location and I think now is not anymore breaking change :)

File: src/Umbraco.Examine/IUmbracoIndexConfig.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using Examine;
+
+namespace Umbraco.Examine
+{
+    public abstract class IUmbracoIndexConfig
```
Comment (Shazwazza) [inline src/Umbraco.Examine/IUmbracoIndexConfig.cs]: A class shouldn't not be prefixed with "I", that is reserved for interfaces. Seems like this should be an interface instead?

File: src/Umbraco.Examine/UmbracoUmbracoIndexConfig.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+using Examine;
+using Umbraco.Core.Services;
+using Umbraco.Core.Services.Implement;
+
+namespace Umbraco.Examine
+{
+    public class UmbracoUmbracoIndexConfig : IUmbracoIndexConfig
```
Comment (Shazwazza) [inline src/Umbraco.Examine/UmbracoUmbracoIndexConfig.cs]: This name has a double prefix of "Umbraco", it should only have one

File: src/Umbraco.Web/Search/UmbracoIndexesCreator.cs [hunk 3]
```diff
@@ -80,28 +84,31 @@ private IIndex CreateMemberIndex()
                 CreateFileSystemLuceneDirectory(Constants.UmbracoIndexes.MembersIndexPath),
                 new CultureInvariantWhitespaceAnalyzer(),
                 ProfilingLogger,
-                GetMemberValueSetValidator());
+                GetMemberValueSetValidator()
+                );
             return index;
         }
-        
+        [Obsolete]
```
Comment (Shazwazza) [inline src/Umbraco.Web/Search/UmbracoIndexesCreator.cs]: We need to have a description of why this is obsoleted in the ctor overload of this attribute, same goes for the other obsoleted ones.

**Comment by bielu** [comment]
@Shazwazza I changed code based on your comments :)

**Comment by Shazwazza** [comment]
Looks good, thanks! will merge.

**Comment by nul800sebastiaan** [comment]
Hi there @bielu,

First of all: **A big #H5YR** for contributing to Umbraco during Hacktoberfest! We are very thankful for the huge amount of PRs submitted, and all the amazing work you've been doing 🥇 

We have had another record breaking amount of PR's in both the CMS, Documentation, and other smaller public repositories. You can read all about the numbers in the [Umbraco Hacktoberfest round-up blog post](https://umbraco.com/blog/pr-team-update-november-2019-hacktoberfest-round-up/).

Due to the amazing work you and others in the community have been doing, we've had a bit of a hard time keeping up. 😅 While all of the PRs for Hacktoberfest might not have been merged yet, you still qualify for receiving some Umbraco swag, congratulations! 🎉 

In the spirit of Hacktoberfest we've prepared some **exclusive Umbraco swag** for all our contributors - _including you_! 

[Receive your swag!](https://docs.google.com/forms/d/e/1FAIpQLSc-K97Zl5k86cVebKWWqieLM4Fm_jCMGVdbQdYnq4IUJ7EvZg/viewform) :point_left: Please follow this link to fill out and submit the form, **before December 19th 2019**.

Following this date we'll be sending out all the swag, which also means that it might not reach your doorstep before February, so please bear with us and be patient :pray:

The only thing left to say is **thank you so much for participating in Hacktoberfest**! We really appreciate the help!

Kind regards,
The Umbraco PR Team & The Documentation Curators

**Pull request merged**